engine/src/main/java/io/camunda/zeebe/engine/processing/common/ElementActivationBehavior.java

```diff
@@ -119,9 +116,6 @@ public ActivatedElementKeys activateElement(
     createVariablesCallback.accept(elementToActivate.getId(), elementInstanceKey);
     activatedElementKeys.setElementInstanceKey(elementInstanceKey);
 
-    // applying the side effects is part of creating the event subscriptions
-    sideEffectQueue.flush();
-
```

Comment:
> But why did that have to happen here? Was there really some use for this, or not?

Glad I'm not the only one confused by this. :sweat_smile: 
I eventually came to the conclusion that this:
1. wasn't necessary because, as you said, the ProcessingStateMachine takes care of flushing side effects
2. was probably a left over (or a copy-paste of a left over) from long ago

Reply:
>it seems dangerous to me that the engine actually flushed side effects on it's own, before the responsible records were even written to the log.

I agree. The solution in this pull request seems much more controlled.